optaplanner-docs/src/main/asciidoc/ConstructionHeuristics/ConstructionHeuristics-chapter.adoc

```diff
@@ -758,14 +758,14 @@ Scales better, at the cost of solution quality. The order of the planning variab
 
 The second way scales better, so it can be worth to switch to it.
 For example, in a course scheduling example with 200 rooms and 40 periods,
-a cartesian product selects 8000 moves per entity (1 step per entity).
+a cartesian product selects 8{nbsp}000 moves per entity (1 step per entity).
 On the other hand, a sequential approach only selects 240 moves per entity (2 steps per entity),
 ending the Construction Heuristic 3 times faster.
 Especially for three or more planning variables, the scaling difference is huge.
-For example, with three variables of 1000 values each,
-a cartesian product selects 1000000000 moves per entity (1 step per entity).
-A sequential approach only selects 240 moves per entity (3 steps per entity),
-ending the Construction Heuristic 300 000 times faster.
+For example, with three variables of 1{nbsp}000 values each,
+a cartesian product selects 1{nbsp}000{nbsp}000{nbsp}000 moves per entity (1 step per entity).
+A sequential approach only selects 3{nbsp}000 moves per entity (3 steps per entity),
+ending the Construction Heuristic 300{nbsp}000 times faster.
```

Comment:
This will definitely improve readability but shouldn't we use the en-US digit grouping style that uses a comma instead of a short space? E.g. 1,000,000,000.

Reply:
It's an interesting point. If the article follows a locale, it is indeed en_US (so "optimizer", not "optimiser" as in en_UK).
But the numbers are following the engineering notation currently: https://en.wikipedia.org/wiki/Engineering_notation, so "1 000 000.00". Dates follow the ISO notation: 2001-9-11 (instead of the very confusing 9/11/2001 which can also mean 9 november for many readers).

Reply:
Hmm, wikepedia says for English: "Digits should be grouped and separated either by commas or by narrow gaps (never a period/full point)"
https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Dates_and_numbers
So no decision. I find gaps much more readable for international audiences, so let's stick with that (but you do have a good point, was worth bringing this up).